### PR TITLE
Add ping/status check to endpoint behavior

### PR DESCRIPTION
**Pull request by k-army**
As discussed in the November quarterly meeting, Caliper sensors should be able to confirm the status of an endpoint by sending it a ping, and getting an "available and ready" response back.

### CODE REVIEW

File: caliper-spec.md [hunk 1]
```diff
@@ -1230,6 +1232,8 @@ A Caliper [Sensor](#sensor) MUST be capable of transmitting Caliper data success
 
 Each message request MUST consist of a single JSON representation of a Caliper [Envelope](#envelope).  Messages MUST be sent using the POST request method.  The HTTP `Host` and `Content-Type` request header fields MUST be set.  
 
+A [Sensor](#sensor) MAY choose to send a ping to an [Endpoint](#endpoint), to confirm that it is available and ready to accept messages, through an empty HTTP GET request.
```
Comment (karaboldt) [inline caliper-spec.md]: Not sure why my IDE decided to strip trailing whitespaces elsewhere, but the relevant changes to review begin here.

File: caliper-spec.md [hunk 2]
```diff
@@ -1240,30 +1244,59 @@ A [Sensor](#sensor) SHOULD also set the `Authorization` request header field usi
...
+| :------- | :--- | ----------- | :---------: |
+| versions | array | An array of the one or more Caliper spec version(s) that the [Endpoint](#endpoint) supports. | Required |
+| maxPayloadSize | integer | The number (in bytes) of the largest payload that the [Endpoint](#endpoint) can accept in one request. | Optional |
+| extensions | object | A map of additional attributes not defined by the model MAY be specified for the [Endpoint](#endpoint). | Optional |
+
+##### Example: Ping response with extensions
+
+```
+{
+  "versions": [ "1.1" ],
```
Comment (ViktorHaag) [inline caliper-spec.md]: Let's use the Caliper context documents supported here -- those are the things we use for versioning. For example, the list could look like this:
```
[
        "http://purl.imsglobal.org/ctx/caliper/v1p1",
        "http://purl.imsglobal.org/ctx/caliper/v1p1/ToolLaunchProfile-extension"
]
```

This lets the response signal that the endpoint supports:
- Caliper v1p1 core
- Tool Launch Profile extensions to v1p1 core

Comment (ViktorHaag) [reply caliper-spec.md]: Follow-up comment -- using the context document URIs to indicate versions is the approach I took in the LTI Caliper Endpoint Service proposal, so whatever approach we take, we should ensure we take the same approach in both places.

Comment (karaboldt) [reply caliper-spec.md]: I agree on the alignment -- was out last week, but I'll update to using the context documents today.

**Comment by k-army** [comment]
Going to close this PR and re-do from the current state of work

**Pull request closed without merging**